drivers/tests/TESTS/mbed_drivers/ticker/main.cpp

```diff
@@ -258,15 +271,16 @@ void test_attach_time(void)
 {
     Ticker ticker;
     ticker_callback_flag = 0;
+    milliseconds delay = milliseconds{DELAY_US};
 
     gtimer.reset();
     gtimer.start();
-    ticker.attach(callback(stop_gtimer_set_flag), ((float)DELAY_US) / 1000000.0f);
+    ticker.attach(callback(stop_gtimer_set_flag), delay);
```

Comment:
Is this correct ? previous value was microseconds divided by 1000000.0 which is presumably converting to seconds. Your delay conversion above is just doing a direct conversion to milliseconds . Is ticker.attach() thus expecting seconds or milliseconds ? Or are there overloading options to cope with both?

Reply:
Chrono types perform resolution conversion - implicitly as long as it's lossless, or explicitly via `duration_cast`. You can't mismatch the periods (except by using the `count()` methods or integer constructors so you're going via integers and mismatching there).

In this case it's passing milliseconds to a microseconds function, so there's an implicit *1000 operation happening.